src/transaction.rs

```diff
@@ -475,19 +475,14 @@ impl<T: Transport> BuildFuture<T> {
                         }
                     )
                     .0
-                    .into_inner()
                 );
 
-                let gas_price = maybe!(builder.gas_price, eth.gas_price());
-                let nonce = maybe!(builder.nonce, eth.transaction_count(from, None));
-
-                // it looks like web3 defaults chain ID to network ID, although
-                // this is not 'correct' in all cases it does work for most cases
-                // like mainnet and various testnets and provides better safety
-                // against replay attacks then just using no chain ID; so lets
-                // reproduce that behaviour here
-                // TODO(nlordell): don't convert to and from string here
-                let chain_id = maybe!(chain_id.map(|id| id.to_string()), net.version());
+                let gas_price = maybe!(builder.gas_price, eth.gas_price().compat());
+                let nonce = maybe!(builder.nonce, eth.transaction_count(from, None).compat());
+                let chain_id = maybe!(
+                    chain_id.map(U64::from),
+                    CallFuture::new(transport.execute("eth_chainId", vec![])).compat()
```

Comment:
Is this a feature released with the recent Istanbul update?

Reply:
https://eips.ethereum.org/EIPS/eip-695